File: dev-docs/snippets/groups/list-databases.ts
import type { JsBaoClient } from "js-bao-wss-client";

// List every database the group has access to, with the granted permission.
export async function listDatabases(
  client: JsBaoClient,
  groupType: string,
  groupId: string,
) {
  // #region example
  const dbs = await client.groups.listDatabases(groupType, groupId);
  for (const db of dbs) {
    console.log(db.databaseId, db.title, db.permission);
  }
  // #endregion example
  return dbs;
}

File: dev-docs/snippets/groups/list-databases.swift
import JsBaoClient

// List every database the group has access to, with the granted permission.
// Returns typed `GroupDatabaseInfo` rows.
func listDatabases(client: JsBaoClient, groupType: String, groupId: String) async throws {
  // #region example
  let dbs = try await client.groups.listDatabases(groupType: groupType, groupId: groupId)
  for db in dbs {
    print(db.databaseId, db.title, db.permission)
  }
  // #endregion example
  _ = dbs
}
